src/modules/hip/kernel/flip_voxel.hpp

```diff
@@ -0,0 +1,188 @@
+#include <hip/hip_runtime.h>
+#include "rpp_hip_common.hpp"
+
+template <typename T>
+__global__ void flip_ncdhw_tensor(T *srcPtr,
+                                  uint3 srcStridesCDH,
+                                  T *dstPtr,
+                                  uint3 dstStridesCDH,
+                                  int channels,
+                                  uint3 mirrorXYZ,
+                                  RpptROI3DPtr roiGenericPtrSrc,
+                                  int batchIndex)
+{
+    int id_x = (hipBlockIdx_x * hipBlockDim_x + hipThreadIdx_x) * 8;        // W - inner most dim vectorized
+    int id_y = hipBlockIdx_y * hipBlockDim_y + hipThreadIdx_y;              // H - second to inner
+    int id_z = hipBlockIdx_z * hipBlockDim_z + hipThreadIdx_z;              // D - outer most dim
+
+    if ((id_z >= roiGenericPtrSrc->xyzwhdROI.roiDepth) || (id_y >= roiGenericPtrSrc->xyzwhdROI.roiHeight) || (id_x >= roiGenericPtrSrc->xyzwhdROI.roiWidth))
```

Comment:
we are assuming the roi always start from 0 for all dims. Is there a case otherwise?

Reply:
Hi Rajy,
we are not assuming ROI to start from 0 for all dims. we are using roiWidth, roiHeight, roiDepth as the boundary conditions, which is the size of output

we are using roi, x, y, z co-ordinates here for loading the corresponding input location
https://github.com/r-abishek/rpp/blob/ar/flip_voxel/src/modules/hip/kernel/flip_voxel.hpp#L24